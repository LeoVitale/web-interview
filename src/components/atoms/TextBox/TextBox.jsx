import React from 'react';
import styles from './TextBox.module.scss';

const TextBox = ({ name, label, ...props }) => {
  const { textBox, title, input } = styles;
  return (
    <div className={textBox}>
      {label && <label className={title} htmlFor={name}>{label}</label>}
      <input className={input} name={name} id={name} {...props} type="text"/>
    </div>
  );
};

export default TextBox;
