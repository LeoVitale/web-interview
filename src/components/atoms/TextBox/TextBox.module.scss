@import 'styles/_constants.scss';

.textBox {
  width: 100%;
  .title{
    display: block;
    font-size: 1.2em;
    margin-bottom: 5px;
  }
  .input {
    width: 100%;
    font-size: 1.4em;
    padding: $gutter-size;
    border: $border-standard;
    border-radius: 4px;
    &:focus {
      border: 1px solid $transparent-grey;
    }
  }
}
