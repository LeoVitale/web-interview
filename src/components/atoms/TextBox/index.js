export { default } from './TextBox.jsx'
