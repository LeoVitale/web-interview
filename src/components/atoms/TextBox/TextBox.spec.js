import React from 'react';
import { mount } from 'enzyme';
import TextBox from './TextBox.jsx';

describe('<TextArea />', () => {
  const textBox = mount(<TextBox />);

  it('should render TextBox as expected', () => {
    expect(textBox).toMatchSnapshot();
  });
});
