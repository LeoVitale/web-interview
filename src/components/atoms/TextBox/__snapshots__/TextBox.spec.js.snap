// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> should render TextBox as expected 1`] = `
<TextBox>
  <div
    className="textBox"
  >
    <input
      className="input"
      type="text"
    />
  </div>
</TextBox>
`;
